### PR TITLE
Refine input validation for GetLocalFunctionControlFlowGraph/GetAnonymousFunctionControlFlowGraph APIs

### PR DESCRIPTION
**Pull request by AlekseyTs**

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@333fred, @mavasani, @cston, @dotnet/roslyn-compiler, @dotnet/analyzer-ioperation Please review

**Review by 333fred — DISMISSED** [review]
Done review pass (Iteration 1)

File: src/Compilers/Core/Portable/Operations/ControlFlowGraph.cs
Comment (AlekseyTs) [reply src/Compilers/Core/Portable/Operations/ControlFlowGraph.cs]: > I don't see tests covering the invalid argument cases?

What about the only test added in this PR, LambdaFlow_04?

---
In reply to: [193931270](https://github.com/dotnet/roslyn/pull/27604#discussion_r193931270) [](ancestors = 193931270)

**Review by 333fred — APPROVED** [review]

File: src/Compilers/Core/Portable/Operations/ControlFlowGraph.cs
Comment (333fred) [reply src/Compilers/Core/Portable/Operations/ControlFlowGraph.cs]: I must have missed the assert nulls. It would be good to break those out into separate tests though, as that test is probably testing too many things at the same time.

**Comment by AlekseyTs** [comment]
@cston, @dotnet/roslyn-compiler Please review, need a second review for a very small change

**Comment by AlekseyTs** [comment]
@mavasani, @dotnet/analyzer-ioperation Please review

**Review by cston — APPROVED** [review]

**Review by mavasani — APPROVED** [review]

**Pull request merged**
